jwcrypto/jwk.py

```diff
@@ -252,11 +252,12 @@ def _generate_RSA(self, params):
             size = params['size']
             del params['size']
         key = rsa.generate_private_key(pubexp, size, default_backend())
-        self._import_pyca_pri_rsa(key)
+        self._import_pyca_pri_rsa(key, **params)
 
-    def _import_pyca_pri_rsa(self, key):
+    def _import_pyca_pri_rsa(self, key, **kwargs):
         pn = key.private_numbers()
-        params = {'kty': 'RSA'}
+        params = kwargs.copy()
```

Comment:
You don't have to copy kwargs. It's already a shallow copy:

>>> d = {'egg': 'spam'}
>>> def func(**kwargs):
...     kwargs['egg'] = 'ham'
...     kwargs['foo'] = 'bar'
...     print(kwargs)
... 
>>> d
{'egg': 'spam'}
>>> func(**d)
{'foo': 'bar', 'egg': 'ham'}
>>> d
{'egg': 'spam'}